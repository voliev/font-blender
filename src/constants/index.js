// *** ACTION TYPES ***

// Background
export const SET_BACKGROUND_COLOR = "SET_BACKGROUND_COLOR";

// Filteres
export const SET_CATEGORIES_FILTER = "SET_CATEGORIES_FILTER";
export const SET_SORTING_ORDER = "SET_SORTING_ORDER";
export const SET_SORTING_PARAM = "SET_SORTING_PARAM";
export const SET_TITLE_FILTER = "SET_TITLE_FILTER";

// Fonts
export const FETCH_FONTS_DATA = "FETCH_FONTS_DATA";
export const FETCH_FONTS_DATA_SUCCESS = "FETCH_FONTS_DATA_SUCCESS";
export const FETCH_FONTS_DATA_FAILURE = "FETCH_FONTS_DATA_FAILURE";
export const SET_FONTS = "SET_FONTS";
export const UPDATE_FONT = "UPDATE_FONT";

// Notifications
export const ADD_NOTIFICATION = "ADD_NOTIFICATION";
export const REMOVE_NOTIFICATION = "REMOVE_NOTIFICATION";

// Showcase
export const DISPLAY_SHOWCASE_TEXT = "DISPLAY_SHOWCASE_TEXT";
export const DISPLAY_STYLES = "DISPLAY_STYLES";
export const DISPLAY_SHOWCASE_FONT_PREVIEW = "DISPLAY_SHOWCASE_FONT_PREVIEW";

// Styles
export const SET_DEFAULT_STATE = "SET_DEFAULT_STATE";
export const UNDO_RESET = "UNDO_RESET";

// Text styles
export const SET_STYLES_INHERITANCE = "SET_STYLES_INHERITANCE";
export const UPDATE_ELEMENT_STYLES = "UPDATE_ELEMENT_STYLES";

// Selected fonts
export const ADD_SELECTED_FONT = "ADD_SELECTED_FONT";
export const CLEAR_SELECTED_FONTS = "CLEAR_SELECTED_FONTS";
export const REMOVE_FONT_FROM_SELECTED = "REMOVE_FONT_FROM_SELECTED";
export const UNDO_DESELECTED = "UNDO_DESELECTED";

// Type scale
export const SET_DEFAULT_TYPE_SCALE_STATE = "SET_DEFAULT_TYPE_SCALE_STATE";
export const SET_FONT_SIZE = "SET_FONT_SIZE";
export const SET_TYPE_SCALE = "SET_TYPE_SCALE";

// *** MISCELLANEOUS ***

// Google Fonts API endpoint URL
export const GOOGLE_FONTS_API_ENDPOINT =
  "https://www.googleapis.com/webfonts/v1/webfonts";
// Fonts' bundle size that should be prefetched
export const BUNDLE_SIZE = 50;
// Available text elements for styling
export const ELEMENTS = ["body", "h1", "h2", "h3", "h4", "li", "small"];
// Font's character range that needs to be prefetched
export const CHAR_RANGE =
  "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
// Font showcase string
export const SHOWCASE_STRING =
  "That's the beauty of music. They can't get that from you...";
// Default showcase font size
export const DEFAULT_FONT_SIZE = 16;
// Available type scale values
export const SCALE_VALUES = {
  1.067: "Minor Second",
  1.125: "Major Second",
  1.2: "Minor Third",
  1.25: "Major Third",
  1.333: "Perfect Fourth",
  1.414: "Augmented Fourth",
  1.5: "Perfect Fifth",
  1.618: "Golden Ratio"
};
// Minimum gap between the color picker icon and viewport
// edge required to place the color picker
export const MIN_GAP = 300;
// Notification snackbar visibility timeout
export const TIMEOUT = 10000;
