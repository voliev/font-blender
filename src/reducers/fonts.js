import { combineReducers } from "redux";
import {
  FETCH_FONTS_DATA,
  FETCH_FONTS_DATA_FAILURE,
  FETCH_FONTS_DATA_SUCCESS,
  SET_FONTS,
  UPDATE_FONT
} from "../constants";

export const fontsDataStatus = (state = "", action) => {
  switch (action.type) {
    case FETCH_FONTS_DATA:
      return "fetching";
    case FETCH_FONTS_DATA_SUCCESS:
      return "success";
    case FETCH_FONTS_DATA_FAILURE:
      return "failure";
    default:
      return state;
  }
};

const initialState = {};

export const all = (state = initialState, action) => {
  switch (action.type) {
    case SET_FONTS:
      return action.fonts;
    case UPDATE_FONT:
      return {
        ...state,
        [action.family]: {
          ...state[action.family],
          ...action.updates
        }
      };
    default:
      return state;
  }
};

export default combineReducers({
  all,
  fontsDataStatus
});

export const getFonts = state => state.fonts.all;
export const getFont = (state, family) => state.fonts.all[family];
export const getFontsDataStatus = state => state.fonts.fontsDataStatus;
