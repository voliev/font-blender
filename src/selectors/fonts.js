import isEqual from "lodash.isequal";

export default function() {
  const memo = {
    filters: {},
    result: []
  };

  return function(fonts, filters) {
    const families = Object.keys(fonts);

    if (families.length === 0) {
      memo.filters = { ...filters };
      memo.result = [];

      return [];
    }

    if (
      (memo.result.length === 0 && families.length > 0) ||
      !isEqual(memo.filters, filters)
    ) {
      memo.filters = { ...filters };

      const { title, categories, sorting } = filters;

      const filteredFonts = families.filter(family => {
        const { category } = fonts[family];

        const titleMatch = family
          .toLowerCase()
          .includes(title.trim().toLowerCase());
        const categoryMatch = categories[category];

        return titleMatch && categoryMatch;
      });

      if (sorting.param === "alpha") {
        const sortedFonts = filteredFonts.sort((a, b) =>
          a.localeCompare(b, {}, { caseFirst: "upper" })
        );

        memo.result =
          sorting.order === "asc" ? sortedFonts : sortedFonts.reverse();
      } else {
        memo.result =
          sorting.order === "asc" ? filteredFonts : filteredFonts.reverse();
      }
    }

    return memo.result;
  };
}
