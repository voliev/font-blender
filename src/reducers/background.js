import { SET_BACKGROUND_COLOR, SET_DEFAULT_STATE } from "../constants";

const initialState = {
  color: "#fff"
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_BACKGROUND_COLOR:
      return {
        ...state,
        color: action.color
      };
    case SET_DEFAULT_STATE:
      return initialState;
    default:
      return state;
  }
};

export const getBackground = state => state.styles.present.background;
export const getBackgroundDefaultState = () => ({ ...initialState });
