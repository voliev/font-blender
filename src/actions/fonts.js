import WebFont from "webfontloader";
import { getFonts } from "../reducers/fonts";
import {
  GOOGLE_FONTS_API_ENDPOINT,
  FETCH_FONTS_DATA,
  FETCH_FONTS_DATA_FAILURE,
  FETCH_FONTS_DATA_SUCCESS,
  SET_FONTS,
  UPDATE_FONT
} from "../constants";

export const fetchFontsData = () => ({
  type: FETCH_FONTS_DATA
});

export const fetchFontsDataSuccess = () => ({
  type: FETCH_FONTS_DATA_SUCCESS
});

export const fetchFontsDataFailure = () => ({
  type: FETCH_FONTS_DATA_FAILURE
});

export const setFonts = fonts => ({
  type: SET_FONTS,
  fonts
});

export const startFetchFontsData = () => {
  return async dispatch => {
    const apiRequestURL = `${GOOGLE_FONTS_API_ENDPOINT}?key=${
      process.env.REACT_APP_GOOGLE_FONTS_API_KEY
    }&sort=popularity`;

    dispatch(fetchFontsData());

    try {
      const res = await fetch(apiRequestURL);
      const { items } = await res.json();
      dispatch(fetchFontsDataSuccess());

      const fonts = items.reduce((result, font) => {
        result[font.family] = {
          family: font.family,
          category: font.category,
          variants: font.variants,
          subsets: font.subsets,
          fetching: false,
          active: false,
          requested: false,
          prefetched: false,
          fetched: false
        };

        return result;
      }, {});

      return dispatch(setFonts(fonts));
    } catch (e) {
      dispatch(fetchFontsDataFailure());
      // TODO: Eject and replace service worker with custom one.
      //       Implement fonts' json data caching.
      return dispatch(setFonts({}));
    }
  };
};

export const updateFont = (family, updates) => ({
  type: UPDATE_FONT,
  family,
  updates
});

export const startFetchFont = (family, charRange = undefined) => {
  return (dispatch, getState) => {
    const state = getState();
    const fonts = getFonts(state);
    const font = fonts[family];
    const fetchFull = !charRange;
    const requestString = generateRequestString(font, fetchFull);
    const options = generateRequestOptions(
      requestString,
      charRange,
      dispatch,
      state
    );

    WebFont.load(options);

    const updatedState = getState();
    const updatedFonts = getFonts(updatedState);

    // For "then" chaining in selectedFonts.js async action creator
    return Promise.resolve(updatedFonts[family]);
  };
};

export const generateRequestString = (font, fetchFull = false) => {
  const { family, subsets, variants } = font;

  if (fetchFull) {
    return `${family}:${variants.join(",")}`;
  }

  const weight = variants.includes("regular")
    ? "400"
    : parseInt(variants[0], 10).toString();
  const subset = subsets.includes("latin") ? "latin" : subsets[0];

  return `${family}:${weight}:${subset}`;
};

export const generateRequestOptions = (request, chars, dispatch) => ({
  google: {
    families: [request],
    text: chars
  },
  classes: false,
  timeout: 10000,
  fontloading: getEventListener("fontloading", dispatch, chars),
  fontactive: getEventListener("fontactive", dispatch, chars),
  fontinactive: getEventListener("fontinactive", dispatch, chars)
});

export const getEventListener = (event, dispatch, charRange) => {
  const listeners = {
    fontloading: family => {
      const updates = Object.assign(
        { fetching: true },
        charRange ? { requested: true } : {}
      );
      dispatch(updateFont(family, updates));
    },
    fontactive: family => {
      const updates = Object.assign(
        { active: true, fetching: false },
        charRange ? { prefetched: true } : { fetched: true }
      );
      dispatch(updateFont(family, updates));
    },
    fontinactive: family => {
      const updates = Object.assign(
        { fetching: false },
        charRange ? { active: false } : {}
      );
      dispatch(updateFont(family, updates));
    }
  };

  return listeners[event];
};
