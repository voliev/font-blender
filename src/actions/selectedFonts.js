import {
  ADD_SELECTED_FONT,
  CLEAR_SELECTED_FONTS,
  REMOVE_FONT_FROM_SELECTED,
  UNDO_DESELECTED
} from "../constants";
import { startFetchFont } from "./fonts";
import { addNotification, removeNotification } from "./notifications";
import { getFonts } from "../reducers/fonts";

export const addFontToSelected = family => ({
  type: ADD_SELECTED_FONT,
  family
});

export const removeFontFromSelected = family => ({
  type: REMOVE_FONT_FROM_SELECTED,
  family
});

export const clearSelectedFonts = () => ({
  type: CLEAR_SELECTED_FONTS
});

export const startAddFontToSelected = family => {
  return (dispatch, getState) => {
    const state = getState();
    const fonts = getFonts(state);
    const font = fonts[family];

    if (font.fetched) {
      // Add font to selected if it has been already fetchedselected list
      dispatch(addFontToSelected(family));
    } else {
      // Fetch font first than add it to selected
      return dispatch(startFetchFont(family))
        .then(() => dispatch(addFontToSelected(family)))
        .catch(() => {});
    }
  };
};

export const undoDeselected = () => ({
  type: UNDO_DESELECTED
});

export const startRemoveFontFromSelected = family => {
  return dispatch => {
    dispatch(removeFontFromSelected(family));

    const text = `${family} font family has been removed`;
    const group = "selectedFonts";
    // Add notification to display it with "undo" option
    dispatch(addNotification(text, group));
  };
};

export const startClearSelectedFonts = () => {
  return (dispatch, getState) => {
    const state = getState();
    const selectedFonts = [...state.selectedFonts.present];

    dispatch(clearSelectedFonts());

    const text =
      selectedFonts.length > 1
        ? `${selectedFonts.length} font families have been removed`
        : `${selectedFonts.length} font family have been removed`;
    const group = "selectedFonts";
    // Add notification to display it with "undo" option
    dispatch(addNotification(text, group));
  };
};

export const startUndoDeselected = () => {
  return dispatch => {
    // Undo deselection
    dispatch(undoDeselected());

    const group = "selectedFonts";

    // Remove notification to hide "undo" option
    dispatch(removeNotification(group));
  };
};
