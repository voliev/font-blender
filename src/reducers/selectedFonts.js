import undoable, { includeAction } from "redux-undo";
import {
  ADD_SELECTED_FONT,
  CLEAR_SELECTED_FONTS,
  REMOVE_FONT_FROM_SELECTED,
  UNDO_DESELECTED
} from "../constants";
import { getFonts } from "./fonts";

const initialState = [];

export const selectedFonts = (state = initialState, action) => {
  switch (action.type) {
    case ADD_SELECTED_FONT:
      return [...state, action.family];
    case REMOVE_FONT_FROM_SELECTED:
      return state.filter(family => family !== action.family);
    case CLEAR_SELECTED_FONTS:
      return [];
    default:
      return state;
  }
};

export default undoable(selectedFonts, {
  limit: 1,
  filter: includeAction([REMOVE_FONT_FROM_SELECTED, CLEAR_SELECTED_FONTS]),
  undoType: UNDO_DESELECTED
});

export const getSelectedFontFamilies = state => state.selectedFonts.present;
export const getSelectedFonts = state => {
  const fonts = getFonts(state);

  return getSelectedFontFamilies(state).reduce((selected, family) => {
    selected[family] = fonts[family];
    return selected;
  }, {});
};
export const isSelected = (state, family) =>
  state.selectedFonts.present.includes(family);
