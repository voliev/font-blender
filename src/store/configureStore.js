import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import fonts from "../reducers/fonts";
import filters from "../reducers/filters";
import selectedFonts from "../reducers/selectedFonts";
import showcase from "../reducers/showcase";
import styles from "../reducers/styles";
import notifications from "../reducers/notifications";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default () => {
  const store = createStore(
    combineReducers({
      fonts,
      filters,
      selectedFonts,
      showcase,
      styles,
      notifications
    }),
    composeEnhancers(applyMiddleware(thunk))
  );

  return store;
};
