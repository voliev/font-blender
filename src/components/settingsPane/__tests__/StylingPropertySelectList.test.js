import React from "react";
import { mount, shallow } from "enzyme";
import configureStore from "redux-mock-store";
import StylingPropertySelectListConnected, {
  StylingPropertySelectList
} from "../StylingPropertySelectList";
import { getSelectedFonts } from "../../../reducers/selectedFonts";
import { UPDATE_ELEMENT_STYLES } from "../../../constants";
import selectedFonts from "../../../fixtures/selectedFonts";
import fonts from "../../../fixtures/fonts";
import state from "../../../fixtures/state";

const setup = (
  renderFn = shallow,
  property = "font-family",
  propsOverrides = {},
  connected = false
) => {
  const options = {
    "font-family": selectedFonts,
    "font-style": ["normal", "italic"],
    "font-weight": [100, 300, 400, 700, 900]
  };
  const value = {
    "font-family": "Lato",
    "font-style": "italic",
    "font-weight": 300
  };
  const props = {
    disabled: false,
    element: "h1",
    selectedFonts: getSelectedFonts(state),
    options: options[property],
    property,
    value: value[property],
    updateElementStyles: jest.fn(),
    ...propsOverrides
  };

  if (connected) {
    const mockStore = configureStore();
    const store = mockStore(state);

    return {
      ...props,
      store,
      wrapper: renderFn(
        <StylingPropertySelectListConnected store={store} {...props} />
      )
    };
  }

  return {
    ...props,
    wrapper: renderFn(<StylingPropertySelectList {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup(shallow, "font-family");

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const {
    disabled,
    options,
    property,
    selectedFonts,
    updateElementStyles,
    value,
    wrapper
  } = setup(mount, "font-style");

  expect(typeof wrapper.prop("disabled")).toBe("boolean");
  expect(wrapper.prop("disabled")).toBe(disabled);

  expect(typeof wrapper.prop("selectedFonts")).toBe("object");
  expect(wrapper.prop("selectedFonts")).toEqual(selectedFonts);

  expect(typeof wrapper.prop("options")).toBe("object");
  expect(wrapper.prop("options").hasOwnProperty("length")).toBeTruthy();
  expect(wrapper.prop("options")).toBe(options);

  expect(typeof wrapper.prop("property")).toBe("string");
  expect(wrapper.prop("property")).toEqual(property);

  expect(typeof wrapper.prop("value")).toBe("string");
  expect(wrapper.prop("value")).toEqual(value);

  expect(typeof wrapper.prop("updateElementStyles")).toBe("function");
  expect(wrapper.prop("updateElementStyles")).toEqual(updateElementStyles);
});

it("should set corresponding component classes", () => {
  const { wrapper } = setup(shallow, "font-weight");

  expect(wrapper.hasClass("StylingPropertySelectList")).toBeTruthy();
  expect(wrapper.hasClass("disabled")).toBeFalsy();

  wrapper.setProps({ disabled: true });

  expect(wrapper.hasClass("StylingPropertySelectList")).toBeTruthy();
  expect(wrapper.hasClass("disabled")).toBeTruthy();
});

it("should handle option change", () => {
  const handleStylePropertyChange = jest.spyOn(
    StylingPropertySelectList.prototype,
    "handleStylePropertyChange"
  );
  const { updateElementStyles, wrapper } = setup(shallow, "font-weight");
  const event = {
    target: {
      value: 700
    }
  };

  wrapper.find("select").simulate("change", event);

  expect(handleStylePropertyChange).toHaveBeenCalledTimes(1);
  expect(updateElementStyles).toHaveBeenCalledTimes(1);
  expect(updateElementStyles).toHaveBeenCalledWith("h1", "font-weight", 700);
});

it("should dispatch an action", () => {
  const { store, wrapper } = setup(shallow, "font-family", undefined, true);

  wrapper.prop("updateElementStyles")("h1", "font-style", "italic");

  const actions = store.getActions();

  expect(actions).toContainEqual({
    type: UPDATE_ELEMENT_STYLES,
    element: "h1",
    property: "font-style",
    value: "italic"
  });
});

it("should set corresponding select list options", () => {
  const { wrapper } = setup(shallow, "font-style", {
    options: undefined,
    value: "normal"
  });

  expect(wrapper.find("option").length).toBe(1);
  expect(wrapper.find("option").text()).toBe("normal");

  const options = ["normal", "italic"];

  wrapper.setProps({ options });

  expect(wrapper.find("option").length).toBe(2);
  wrapper.find("option").forEach((node, idx) => node.text() === options[idx]);
});

it("should return correct value for `font-family` property", () => {
  const getStylePropertyValueFromEvent = jest.spyOn(
    StylingPropertySelectList.prototype,
    "getStylePropertyValueFromEvent"
  );
  const { wrapper } = setup(shallow, "font-family");

  wrapper.instance().getStylePropertyValueFromEvent({
    target: { value: "Lato" }
  });

  expect(getStylePropertyValueFromEvent).toHaveLastReturnedWith(fonts["Lato"]);

  wrapper.instance().getStylePropertyValueFromEvent({
    target: { value: "sans-serif" }
  });

  expect(getStylePropertyValueFromEvent).toHaveLastReturnedWith("sans-serif");
});

it("should return correct value for `font-weight` property", () => {
  const getStylePropertyValueFromEvent = jest.spyOn(
    StylingPropertySelectList.prototype,
    "getStylePropertyValueFromEvent"
  );
  const { wrapper } = setup(shallow, "font-weight");

  wrapper.instance().getStylePropertyValueFromEvent({
    target: { value: "500" }
  });

  expect(getStylePropertyValueFromEvent).toHaveReturnedWith(500);
});

it("should return correct value for `font-style` property", () => {
  const getStylePropertyValueFromEvent = jest.spyOn(
    StylingPropertySelectList.prototype,
    "getStylePropertyValueFromEvent"
  );
  const { wrapper } = setup(shallow, "font-style");

  wrapper.instance().getStylePropertyValueFromEvent({
    target: { value: "italic" }
  });

  expect(getStylePropertyValueFromEvent).toHaveReturnedWith("italic");
});

it("should update component on props change", () => {
  const { wrapper, disabled, value, selectedFonts, options } = setup();
  const shouldComponentUpdate = jest.spyOn(
    StylingPropertySelectList.prototype,
    "shouldComponentUpdate"
  );
  const props = {
    disabled,
    value,
    selectedFonts,
    options
  };
  const propsUpdates = {
    disabled: true,
    value: "Montserrat",
    selectedFonts: {},
    options: []
  };

  for (let prop in props) {
    expect(wrapper.instance().props[prop]).toEqual(props[prop]);

    wrapper.setProps({ [prop]: propsUpdates[prop] });

    expect(wrapper.instance().props[prop]).toEqual(propsUpdates[prop]);
    expect(shouldComponentUpdate).toHaveLastReturnedWith(true);
  }
});

it("should not update component on props change", () => {
  const { wrapper } = setup();
  const shouldComponentUpdate = jest.spyOn(
    StylingPropertySelectList.prototype,
    "shouldComponentUpdate"
  );

  wrapper.setProps({ updateElementStyles: jest.fn() });

  expect(shouldComponentUpdate).toHaveBeenCalledTimes(1);
  expect(shouldComponentUpdate).toHaveLastReturnedWith(false);
});
