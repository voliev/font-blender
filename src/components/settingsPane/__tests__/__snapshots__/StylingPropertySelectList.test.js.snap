// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="StylingPropertySelectList"
>
  <label
    className="label"
    htmlFor="h1-font-family"
  >
    Font family
  </label>
  <select
    className="select-list"
    disabled={false}
    id="h1-font-family"
    name="font-family"
    onChange={[Function]}
    value="Lato"
  >
    <option
      key="default"
      value="default"
    >
      sans-sefir
    </option>
    <option
      key="Lato"
      value="Lato"
    >
      Lato
    </option>
    <option
      key="Montserrat"
      value="Montserrat"
    >
      Montserrat
    </option>
    <option
      key="Roboto Condensed"
      value="Roboto Condensed"
    >
      Roboto Condensed
    </option>
    <option
      key="Raleway"
      value="Raleway"
    >
      Raleway
    </option>
  </select>
</div>
`;
